### ISSUE TITLE
Lua error inside app.transaction causes undefined behavior and crashes.

### ISSUE THREAD

**Issue by nulldisk** [aseprite/aseprite#3276]
If Lua throws an error inside `app.transaction` it is causing undefined behavior, eventually leading to segmentation fault.

It can be as simple as:
```lua
app.transaction(function()
    assert(crash)
end)
```
This causes a lot of random undefined behavior, mostly related to the undo history not working anymore or getting corrupted. Weird things can happen with selections too. If you continue in this state segmentation fault happens eventually.

It is possible to avoid it if you run your code inside a protected call like this:
```lua
app.transaction(function()
    local status, err = pcall(function()
        assert(crash)
    end)

    if status == false then
        print(err)
    end
end)
```
In this case the error is printed and nothing breaks.

I am on Linux, but I've briefly tried this on Windows as well and observed exact same behavior.

### Aseprite and System version

* Aseprite version: 1.3-beta-14-x64 (steam), 1.2.34.1-x64 (steam)
* System: archlinux 5.17.3-arch1-1

**Comment by lampysprites** [comment]
i think it's fixed in 4d18200 - same problem as #4431

**Comment by RefinedSoftwareLLC** [comment]
This is still happening with Aseprite v1.3.6-x64's AppImage.

This is my workaround: 

```
local try = function(label, func)
  if not app.transaction(label,
    function()
      local ok, err = pcall(func)
      if ok then return true end
      app.alert{title=label, text=err}
    end)
  then
    app.undo()
  end
end
```

**Comment by dacap** [comment]
Fixed in v1.3.7 released today with 4d18200b0c6a1c557d4f05f757bc264338104aa9
